File: 백준/Silver/1181. 단어 정렬/단어 정렬.py

word_lst = []
Test = int(input())
for T in range(Test):
    word = input()
    word_lst.append(word)
word_lst = list(set(word_lst))
word_lst.sort()
word_lst.sort(key = lambda x:len(x))
for w in word_lst:
    print(w)

File: 백준/Silver/1181. 단어 정렬/단어 정렬.js
const fs = require('fs');

const inputs = fs
  .readFileSync(0)
  .toString()
  .trim()
  .split('\n')
  .map(input => input.trim());

const alphaSet = new Set(inputs.slice(1));

const alphaArray = [...alphaSet];

const G = Array.from({ length: 51 }, () => []);

for (const alpha of alphaArray) {
  G[alpha.length].push(alpha);
}

for (const alphas of G) {
  if (alphas.length > 0) {
    alphas.sort();
  }
}
let resultArray = [];
for (const alphas of G) {
  resultArray = resultArray.concat(alphas);
}

console.log(resultArray.join('\n'));